### ISSUE TITLE
使用paddle2.2.2时出现的错误When Variable is used as the condition of if/while , Variable can only contain one element.

### ISSUE THREAD

**Issue by shangshanruowo** [PaddlePaddle/Paddle#42441]
### 请提出你的问题 Please ask your question

![1212](https://user-images.githubusercontent.com/101441487/166237577-3430b076-8447-4661-ab77-e146e2085cd5.png)

当我使用这个测试代码时会出现如下错误
![3232](https://user-images.githubusercontent.com/101441487/166237728-9f39d82c-4a05-42eb-9844-61fd830fd5b5.png)
请问是那里的问题，我看别人paddle和pytorch写的都是这样的

**Comment by littletomatodonkey** [comment]
你好，建议贴一段可以复现问题的单测出来看看

**Comment by shangshanruowo** [comment]
> 你好，建议贴一段可以复现问题的单测出来看看
![111](https://user-images.githubusercontent.com/101441487/166395480-f736879c-d5f2-4cdc-9532-691e79d19308.png)
您好，当我实例化对象，并随机创建一个[1,3,640,640]shape的tensor传入网络时就会报如下错误：
![2222](https://user-images.githubusercontent.com/101441487/166395652-a728a127-4c34-42fc-b6d6-2275d11c060a.png)
麻烦您请看看是那里的原因。

**Comment by littletomatodonkey** [comment]
这个可以直接给一下代码片段吗？把代码文字贴出来

**Comment by shangshanruowo** [comment]
import paddle.nn as nn
import paddle 
import numpy as np 
def autopad(k, p=None):
    if p is None:
        p = k // 2 if isinstance(k, int) else [x // 2 for x in k] 
    return p
class Conv(nn.Layer):
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):
        super(Conv, self).__init__()
        self.conv = nn.Conv2D(c1, c2, k, s, autopad(k, p), groups=g, bias_attr=False)
        self.bn = nn.BatchNorm2D(c2, epsilon=0.001, momentum=0.03)
        #self.act = SiLU() if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())
        self.act = nn.Silu if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))

    def fuseforward(self, x):
        return self.act(self.conv(x))
a=Conv(c1=3,c2=64,k=3)
x1 = np.random.randn(1,3,640,640).astype('float32')
x1 = paddle.to_tensor(x1)
a(x1)

**Comment by shangshanruowo** [comment]
`
import paddle.nn as nn
import paddle 
import numpy as np 
def autopad(k, p=None):
    if p is None:
        p = k // 2 if isinstance(k, int) else [x // 2 for x in k] 
    return p
class Conv(nn.Layer):
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):
        super(Conv, self).__init__()
        self.conv = nn.Conv2D(c1, c2, k, s, autopad(k, p), groups=g, bias_attr=False)
        self.bn = nn.BatchNorm2D(c2, epsilon=0.001, momentum=0.03)
        #self.act = SiLU() if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())
        self.act = nn.Silu if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))

    def fuseforward(self, x):
        return self.act(self.conv(x))
a=Conv(c1=3,c2=64,k=3)
x1 = np.random.randn(1,3,640,640).astype('float32')
x1 = paddle.to_tensor(x1)
a(x1)`

**Comment by shangshanruowo** [comment]
`
import paddle.nn as nn
import paddle 
import numpy as np 
def autopad(k, p=None):
    if p is None:
        p = k // 2 if isinstance(k, int) else [x // 2 for x in k] 
    return p
class Conv(nn.Layer):
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):
        super(Conv, self).__init__()
        self.conv = nn.Conv2D(c1, c2, k, s, autopad(k, p), groups=g, bias_attr=False)
        self.bn = nn.BatchNorm2D(c2, epsilon=0.001, momentum=0.03)
        #self.act = SiLU() if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())
        self.act = nn.Silu if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))

    def fuseforward(self, x):
        return self.act(self.conv(x))
a=Conv(c1=3,c2=64,k=3)
x1 = np.random.randn(1,3,640,640).astype('float32')
x1 = paddle.to_tensor(x1)
a(x1)`

**Comment by littletomatodonkey** [comment]
那个\`符号可以改成3个\`\`\`

**Comment by shangshanruowo** [comment]
```
import paddle.nn as nn
import paddle 
import numpy as np 
def autopad(k, p=None):
    if p is None:
        p = k // 2 if isinstance(k, int) else [x // 2 for x in k] 
    return p
class Conv(nn.Layer):
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):
        super(Conv, self).__init__()
        self.conv = nn.Conv2D(c1, c2, k, s, autopad(k, p), groups=g, bias_attr=False)
        self.bn = nn.BatchNorm2D(c2, epsilon=0.001, momentum=0.03)
        #self.act = SiLU() if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())
        self.act = nn.Silu if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())

    def forward(self, x):
        return self.act(self.bn(self.conv(x)))

    def fuseforward(self, x):
        return self.act(self.conv(x))
a=Conv(c1=3,c2=64,k=3)
x1 = np.random.randn(1,3,640,640).astype('float32')
x1 = paddle.to_tensor(x1)
a(x1)
```

**Comment by littletomatodonkey** [comment]
这一行
```
self.act = nn.Silu if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())
```

需要修改为

```
self.act = nn.Silu() if act is True else (act if isinstance(act, nn.Layer) else nn.Identity())
```

**Comment by shangshanruowo** [comment]
好的
